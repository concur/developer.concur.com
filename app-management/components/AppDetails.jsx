/* eslint-env browser */

import React, { PropTypes } from 'react';
import { Link } from 'react-router';

class AppDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      app: {},
      loading: false,
      error: false,
    };
  }

  componentWillMount() {
    window.fetch(`${process.env.API_SERVER}/apps/${this.props.params.id}`)
      .then(response => response.json())
      .then(app => this.setState({
        app,
        loading: false,
      }))
      .catch(() => this.setState({
        error: true,
        loading: false,
      }));
  }

  render () {
    if (this.state.loading) {
      return (
        <div className="row">
          <div className="col-md-12">
            <p>Loading...</p>
          </div>
        </div>
      );
    }

    if (this.state.error) {
      return (
        <div className="row">
          <div className="col-md-12">
            <p>An error occurred when loading this app</p>
            <Link to="/">Back to My Apps</Link>
          </div>
        </div>
      );
    }

    const { app } = this.state;

    return (
      <div className="row">
        <div className="col-md-12">
          <h2>{app.name}</h2>
          <Link to="/">&larr; Back to My Apps</Link>
          <div className="row">
            <p>{app.description}</p>
          </div>
        </div>
      </div>
    );
  }
}

AppDetails.propTypes = {
  params: PropTypes.shape({
    id: PropTypes.string.isRequired,
  }),
};

export default AppDetails;
