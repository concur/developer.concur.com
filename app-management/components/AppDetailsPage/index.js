/* eslint-env browser */

import { connect } from 'react-redux';
import AppDetailsPage from './AppDetailsPage';
import { fetchAppDetails, updateAppDetails, showSecret } from '../../actions/appDetails';

export function mapStateToProps(state) {
  return state.appDetails;
}

export function mapDispatchToProps(dispatch) {
  return {
    fetchAppDetails: id => dispatch(fetchAppDetails(id)),
    handleSubmit: app => dispatch(updateAppDetails(app)),
    generateSecret: (id) => {
      const confirmation = window.confirm('Are you sure? This will clear your previous app secret.');
      if (confirmation) {
        dispatch(showSecret());
        console.log(`TBD: Generate secret for app ${id}`);
      }
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AppDetailsPage);
