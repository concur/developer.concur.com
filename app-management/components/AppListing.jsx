import React from 'react';
import { Link } from 'react-router';

class AppListing extends React.Component {
  constructor(props) {
    super(props);
    const apps = [
      {
        id: '2342asdfas',
        name: 'Starbucks',
      },
      {
        id: '234asdfasdf',
        name: 'Uber',
      },
      {
        id: 'asldfkajsdflj',
        name: 'My App',
      },
    ];
    this.state = { apps };
  }

  render () {
    const appItems = this.state.apps.map((app) => {
      const { id, name } = app;
      return (
        <div className="well" key={id}>
          <h2>{name}</h2>
          <Link to="" className="btn-u btn-u-green">Details</Link>
          <Link to="" className="btn-u btn-u-blue">Edit</Link>
        </div>
      );
    });

    return (
      <div className="row">
        <div className="col-md-12">
          <h1>My Apps</h1>
          <div className="row">
            {appItems}
          </div>
        </div>
      </div>
    );
  }
}

export default AppListing;
