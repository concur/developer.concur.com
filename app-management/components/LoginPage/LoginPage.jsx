import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import LoginForm from './LoginForm';
import LoadingSpinner from '../LoadingSpinner';
import ErrorAlert from '../ErrorAlert';

const LoginPage = ({ handleSubmit, isFetching, error }) => (
  <div className="row">
    <div className="col-md-6">
      <h2>Log in to App Management</h2>
      <p>Don&#39;t have an account? <Link to="/signup">Sign Up here</Link></p>
      <ErrorAlert error={error} />
      <LoadingSpinner loading={isFetching} />
      <LoginForm onSubmit={handleSubmit} />
    </div>
  </div>
);

LoginPage.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  isFetching: PropTypes.bool,
  error: PropTypes.string,
};

export default LoginPage;
