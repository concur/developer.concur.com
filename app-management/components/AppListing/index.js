import { connect } from 'react-redux';
import AppListing from './AppListing';
import { fetchAppListing } from '../../actions/appListing';

export function mapStateToProps(state) {
  return state.appListing;
}

export function mapDispatchToProps(dispatch) {
  return {
    fetchAppListing: () => dispatch(fetchAppListing()),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AppListing);
