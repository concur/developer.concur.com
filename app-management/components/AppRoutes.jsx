import React, { PropTypes } from 'react';
import { Router, Route, IndexRoute, hashHistory } from 'react-router';

import PageContainer from './PageContainer';
import AppListing from './AppListing';
import AppDetailsPage from './AppDetailsPage';
import NewAppPage from './NewAppPage';
import LoginPage from './LoginPage';
import SignupPage from './SignupPage';
import LogoutPage from './LogoutPage';
import NotFound from './NotFound';

import { logout } from '../actions/auth';

class AppRoutes extends React.Component {
  constructor(props) {
    super(props);

    this.isAuthenticated = this.isAuthenticated.bind(this);
    this.requireAuth = this.requireAuth.bind(this);
    this.showAuthForms = this.showAuthForms.bind(this);
    this.logout = this.logout.bind(this);
  }

  isAuthenticated() {
    const { store } = this.props;
    return !!store.getState().auth.token;
  }

  requireAuth(nextState, replaceState) {
    if (!this.isAuthenticated()) {
      replaceState('/login');
    }
  }

  showAuthForms(nextState, replaceState) {
    if (this.isAuthenticated()) {
      replaceState('/');
    }
  }

  logout(nextState, replaceState) {
    if (this.isAuthenticated()) {
      this.props.store.dispatch(logout());
    } else {
      replaceState('/login');
    }
  }

  render () {
    return (
      <Router history={hashHistory}>
        <Route path="/" component={PageContainer}>
          <IndexRoute component={AppListing} onEnter={this.requireAuth} />
          <Route path="new" component={NewAppPage} onEnter={this.requireAuth} />
          <Route path="details/:id" component={AppDetailsPage} onEnter={this.requireAuth} />
          <Route path="login" component={LoginPage} onEnter={this.showAuthForms} />
          <Route path="signup" component={SignupPage} onEnter={this.showAuthForms} />
          <Route path="logout" component={LogoutPage} onEnter={this.logout} />
          <Route path="*" component={NotFound} />
        </Route>
      </Router>
    );
  }
}

AppRoutes.propTypes = {
  store: PropTypes.shape({
    dispatch: PropTypes.func.isRequired,
  }),
};

export default AppRoutes;
