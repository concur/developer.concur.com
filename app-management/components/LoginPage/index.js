import { connect } from 'react-redux';
import LoginPage from './LoginPage';
import { login } from '../../actions/auth';

export function mapStateToProps(state) {
  return state.auth;
}

export function mapDispatchToProps(dispatch) {
  return {
    handleSubmit: user => dispatch(login(user)),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LoginPage);
