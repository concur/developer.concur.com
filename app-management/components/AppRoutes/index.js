import AppRoutes from './AppRoutes';

export default AppRoutes;
