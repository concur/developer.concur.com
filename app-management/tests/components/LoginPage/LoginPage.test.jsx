import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';
import { shallowToJson } from 'enzyme-to-json';

import LoginPage from '../../../components/LoginPage/LoginPage';
import { mapStateToProps, mapDispatchToProps } from '../../../components/LoginPage';
import appReducer from '../../../reducers';

const store = createStore(appReducer);

describe('<LoginPage />', () => {
  it('should render with <LoginForm />', () => {
    const handleSubmit = jest.fn();
    const page = shallow(<LoginPage handleSubmit={handleSubmit} />);

    expect(shallowToJson(page)).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('returns entire auth state', () => {
      const state = mapStateToProps(store.getState());

      expect(state).toEqual(store.getState().auth);
    });
  });

  describe('mapDispatchToProps', () => {
    it('returns submit handler', () => {
      const dispatcher = jest.fn();
      const state = mapDispatchToProps(dispatcher);
      state.handleSubmit('user');

      expect(state.handleSubmit).toBeDefined();
      expect(dispatcher).toHaveBeenCalled();
    });
  });
});
