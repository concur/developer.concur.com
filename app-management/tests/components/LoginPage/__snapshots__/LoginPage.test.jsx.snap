exports[`<LoginPage /> should render with <LoginForm /> 1`] = `
<div
  className="row">
  <div
    className="col-md-6">
    <h2>
      Log in to App Management
    </h2>
    <p>
      Don\'t have an account? 
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/signup">
        Sign Up here
      </Link>
    </p>
    <ErrorAlert />
    <LoadingSpinner />
    <ReduxForm
      onSubmit={[Function]} />
  </div>
</div>
`;
